components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -0,0 +1,2222 @@
+       Project proj = mmFactory.simpleProject();
+       Dataset dat = mmFactory.simpleDataset();
+       Project sentProj = new ProjectI();
+       sentProj = null;
+       Dataset sentDat = new DatasetI();
+       sentDat = null;
+       try {
+           sentProj = (Project) iUpdate.saveAndReturnObject(proj);
+           sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
+       } catch (ServerError se) {
+       }
+       /* import an image for the normalUser into the normalUser's default group 
+        * and target it into the created Dataset*/
+       final RString imageName = omero.rtypes.rstring(fakeImageFile.getName());
+       final List<List<RType>> result = iQuery.projection(
+               "SELECT id FROM OriginalFile WHERE name = :name ORDER BY id DESC LIMIT 1",
+               new ParametersI().add("name", imageName));
+       final long previousId = result.isEmpty() ? -1 : ((RLong) result.get(0).get(0)).getValue();
+       try {
+           List<String> path = Collections.singletonList(fakeImageFile.getPath());
+           importFileset(path, path.size(), sentDat);
+       } catch (ServerError se) {
+       }
+       final List<RType> resultAfterImport = iQuery.projection(
+               "SELECT id, details.group.id FROM OriginalFile o WHERE o.id > :id AND o.name = :name",
+               new ParametersI().addId(previousId).add("name", imageName)).get(0);
+       final long remoteFileId = ((RLong) resultAfterImport.get(0)).getValue();
+       final long remoteFileGroupId = ((RLong) resultAfterImport.get(1)).getValue();
+       assertOwnedBy((new OriginalFileI(remoteFileId, false)), normalUser);
+       Assert.assertEquals(remoteFileGroupId, normalUser.groupId);
+       /* link the Project and the Dataset */
+       ProjectDatasetLink link = linkProjectDataset(sentProj, sentDat);
+       Image image = (Image) iQuery.findByQuery(
+               "FROM Image WHERE fileset IN "
+               + "(SELECT fileset FROM FilesetEntry WHERE originalFile.id = :id)",
+               new ParametersI().addId(remoteFileId));
+       IObject datasetImageLink = iQuery.findByQuery(
+               "FROM DatasetImageLink WHERE child.id = :id",
+               new ParametersI().addId(image.getId()));
+       IObject projectDatasetLink = iQuery.findByQuery(
+               "FROM ProjectDatasetLink WHERE id = :id",
+               new ParametersI().addId(link.getId()));
+       /* take care of post-import workflows which do not use sudo */
+       if (!isSudoing) {
+           loginUser(lightAdmin);
+       }
+       /* Now check that the ImporterAs can delete the objects
+        * created on behalf of the user. Note that deletion of the Project
+        * would delete the whole hierarchy, which was successfully tested
+        * during writing of this test.*/
+       if (deletePassing) {
+           doChange(Requests.delete().target(datasetImageLink).build());
```

Comment:
use `target(IObject... targets)` for all at once? or if that doesn't work, can just do `target(foo).target(bar)...`